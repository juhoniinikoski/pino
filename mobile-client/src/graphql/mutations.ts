import { gql } from '@apollo/client';

export const AUTHORIZE = gql`
  mutation authorize($email: String!, $password: String!) {
    authorize(credentials: { email: $email, password: $password }) {
      accessToken
    }
  }
`;

export const FOLLOW_COLLECTION = gql`
  mutation followCollection($collectionId: ID!) {
    followCollection(collectionId: $collectionId)
  }
`;
