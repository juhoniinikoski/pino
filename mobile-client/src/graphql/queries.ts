import { gql } from '@apollo/client';

export const GET_CHANNELS = gql`
  query Channels($first: Int, $after: String, $searchKeyword: String) {
    channels(first: $first, after: $after, searchKeyword: $searchKeyword) {
      edges {
        node {
          id
          name
          followedBy
          questions
        }
        cursor
      }
      totalCount
      pageInfo {
        endCursor
        startCursor
        hasNextPage
      }
    }
  }
`;

export const GET_FOLLOWED = gql`
  query FollowedCollections(
    $first: Int
    $after: String
    $searchKeyword: String
  ) {
    followedCollections(
      first: $first
      after: $after
      searchKeyword: $searchKeyword
      orderBy: "connectionDate"
    ) {
      edges {
        node {
          ... on Stack {
            id
            name
            followedBy
            questions
            connectionDate
            createdById
          }
          ... on Channel {
            id
            name
            followedBy
            questions
            connectionDate
          }
        }
        cursor
      }
      totalCount
      pageInfo {
        endCursor
        startCursor
        hasNextPage
      }
    }
  }
`;

export const GET_QUESTIONS = gql`
  query Questions(
    $collectionId: ID
    $createdBy: ID
    $first: Int
    $after: String
  ) {
    questions(
      collectionId: $collectionId
      createdBy: $createdBy
      first: $first
      after: $after
    ) {
      edges {
        node {
          id
          question
          createdById
          type
          answers {
            answer
            correct
            id
          }
          updatedAt
          createdAt
        }
        cursor
      }
      totalCount
      pageInfo {
        startCursor
        endCursor
        hasNextPage
        hasPreviousPage
      }
    }
  }
`;

export const GET_AUTHORIZED_USER = gql`
  query AuthorizedUser {
    authorizedUser {
      id
      email
    }
  }
`;
