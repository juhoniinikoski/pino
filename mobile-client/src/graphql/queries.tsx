import { gql } from '@apollo/client';

export const GET_CHANNELS = gql`
  query Channels($first: Int, $after: String, $searchKeyword: String) {
    channels(first: $first, after: $after, searchKeyword: $searchKeyword) {
      edges {
        node {
          id
          name
          followedBy
          questions
        }
        cursor
      }
      totalCount
      pageInfo {
        endCursor
        startCursor
        hasNextPage
      }
    }
  }
`;

export const GET_QUESTIONS = gql`
  query Questions(
    $channelId: ID
    $stackId: ID
    $createdBy: ID
    $first: Int
    $after: String
  ) {
    questions(
      channelId: $channelId
      stackId: $stackId
      createdBy: $createdBy
      first: $first
      after: $after
    ) {
      edges {
        node {
          id
          question
          createdById
          type
          answers {
            answer
            correct
            id
          }
          updatedAt
          createdAt
        }
        cursor
      }
      totalCount
      pageInfo {
        startCursor
        endCursor
        hasNextPage
        hasPreviousPage
      }
    }
  }
`;
