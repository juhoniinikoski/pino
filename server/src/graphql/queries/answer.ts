import { gql } from 'apollo-server';
import { getAnswer } from '../../services/answer/answerService';

export const typeDefs = gql`
  extend type Query {
    """
    Returns an answer.
    """
    answer(id: ID!): Answer!
  }
`;

interface Args {
  id: string | number;
}

export const resolvers = {
  Query: {
    answer: (_obj: null, { id }: Args) => getAnswer(id),
  },
};

export default {
  typeDefs,
  resolvers,
};
