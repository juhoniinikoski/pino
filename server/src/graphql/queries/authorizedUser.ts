import { gql } from 'apollo-server'
import AuthService from '../../services/authentication/authService'
import { getAuthorizedUser } from '../../services/user/userService'
import { Context } from '../../utils/entities'

export const typeDefs = gql`
  extend type Query {
    """
    Returns the authorized user.
    """
    authorizedUser: User
  }
`

export const resolvers = {
  Query: {
    authorizedUser: (_obj: null, _args: null, { authService }: Context) => getAuthorizedUser(authService)
  },
}

export default {
  typeDefs,
  resolvers,
}