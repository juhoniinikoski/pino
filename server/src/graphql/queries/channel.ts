import { gql } from 'apollo-server';
import { getChannel } from '../../services/channel/channelService';

export const typeDefs = gql`
  extend type Query {
    """
    Returns a stack and related questions.
    """
    channel(id: ID!): Channel!
  }
`;

interface Args {
  id: string | number;
}

export const resolvers = {
  Query: {
    stack: async (_obj: null, { id }: Args) => getChannel(id),
  },
};

export default {
  typeDefs,
  resolvers,
};
