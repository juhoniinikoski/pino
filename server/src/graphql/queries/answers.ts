import { gql } from 'apollo-server';
import { getAnswers } from '../../services/answer/answerService';

export const typeDefs = gql`
  extend type Query {
    """
    Returns list of answers.
    """
    answers: [Answer!]
  }
`;

export const resolvers = {
  Query: {
    answers: async () => getAnswers(),
  },
};

export default {
  typeDefs,
  resolvers,
};
