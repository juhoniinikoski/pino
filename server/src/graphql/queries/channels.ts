import { gql } from 'apollo-server';
import { getChannels } from '../../services/channel/channelService';
import { Context } from '../../utils/entities';

export const typeDefs = gql`
  extend type Query {
    """
    Returns list of channels.
    """
    channels(first: Int, after: String, searchKeyword: String, followedByAuthorized: Boolean): ChannelConnection!
  }
`;

interface Args {
  first?: number;
  after?: string;
  orderBy?: string;
  searchKeyword?: string;
  followedByAuthorized?: boolean;
}

export const resolvers = {
  Query: {
    channels: (_obj: null, args: Args, { authService }: Context) => getChannels(args, authService),
  },
};

export default {
  typeDefs,
  resolvers,
};
