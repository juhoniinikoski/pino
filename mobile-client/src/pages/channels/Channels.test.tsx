import * as React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { render, RenderAPI, waitFor } from '@testing-library/react-native';
import { GET_CHANNELS } from '../../graphql/queries';
import ChannelsPage from './Channels';

/* eslint-disable no-unused-expressions */

const mocks = [
  {
    request: {
      query: GET_CHANNELS,
    },
    result: {
      data: [
        {
          cursor:
            'WyIyMDIyLTA0LTI4VDA4OjEyOjQ2LjI0MVoiLCJrYXVwcGlrc2VlbjEyMzQiXQ==',
          node: {
            followedBy: 2,
            id: 'kauppikseen1234',
            name: 'kauppikseen',
            questions: 4,
          },
        },
        {
          cursor: 'WyIyMDIyLTA0LTI4VDA3OjEyOjQ2LjI0MVoiLCJESUEyMDIyMTIzNCJd',
          node: {
            followedBy: 1,
            id: 'DIA20221234',
            name: 'DIA2022',
            questions: 1,
          },
        },
        {
          cursor:
            'WyIyMDIyLTA0LTI4VDA2OjEyOjQ2LjI0MVoiLCJ5b2Z5c2lpa2thMTIzNCJd',
          node: {
            followedBy: 0,
            id: 'yofysiikka1234',
            name: 'yofysiikka',
            questions: 0,
          },
        },
        {
          cursor:
            'WyIyMDIyLTA0LTI4VDA2OjEyOjQ2LjI0MVoiLCJ5b21hdGVtYXRpaWtrYTEyMzQiXQ==',
          node: {
            followedBy: 0,
            id: 'yomatematiikka1234',
            name: 'yomatematiikka',
            questions: 0,
          },
        },
        {
          cursor: 'WyIyMDIyLTA0LTI4VDA1OjEyOjQ2LjI0MVoiLCJ0dXRhMTEyMzQiXQ==',
          node: {
            followedBy: 0,
            id: 'tuta11234',
            name: 'tuta1',
            questions: 0,
          },
        },
      ],
    },
  },
];

describe('rendering tests', () => {
  let component: RenderAPI;

  beforeEach(() => {
    component = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ChannelsPage />
      </MockedProvider>,
    );
  });

  test('renders without error', () => {
    expect(component).toBeDefined;
  });

  test('renders subheaders', async () => {
    expect((await component.findAllByTestId('subheader')).length).toBe(2);
  });

  test('should render lists of channels succesfully', async () => {
    await waitFor(async () => {
      expect(component.getAllByTestId('channels-list').length).toBe(1);
      expect(component).toMatchSnapshot();
    });
  });

  test('should show loading indicator when data is initially loaded', async () => {
    await waitFor(() => {
      expect(component.getByText('Loading')).toBeTruthy;
      expect(component).toMatchSnapshot();
    });
  });
});
