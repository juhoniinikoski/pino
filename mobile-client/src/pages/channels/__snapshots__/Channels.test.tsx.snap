// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering tests should render lists of channels succesfully 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Array [
      Object {
        "display": "flex",
        "flex": 1,
        "marginHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [],
          "renderItem": [Function],
          "title": "Omat kanavat",
        },
        Object {
          "data": Array [],
          "renderItem": [Function],
          "title": "Suositellut kanavat",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginTop": 8,
      }
    }
    testID="channels-list"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 13,
                "fontWeight": "bold",
              },
              Object {
                "marginBottom": 8,
                "marginTop": 16,
              },
            ]
          }
          testID="subheader"
        >
          Omat kanavat
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 13,
                "fontWeight": "bold",
              },
              Object {
                "marginBottom": 8,
                "marginTop": 16,
              },
            ]
          }
          testID="subheader"
        >
          Suositellut kanavat
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`rendering tests should show loading indicator when data is initially loaded 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Array [
      Object {
        "display": "flex",
        "flex": 1,
        "marginHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    Loading
  </Text>
</RNCSafeAreaView>
`;
